storage/database.go

```diff
@@ -292,8 +308,13 @@ func (d *db) FetchBlocks(
 	starts []time.Time,
 ) ([]block.FetchBlockResult, error) {
 	callStart := d.nowFn()
-	n, err := d.namespaceFor(namespace)
+	if d.isOverloaded() {
```

Comment:
Thinking we should move this up a layer and just expose the `IsOverloaded()` method.  That way people building a general application on top of these components don't need to somehow circumvent it if they don't care about the backoff procedure.

i.e. Use this `IsOverloaded()` check before even beginning an RPC and making the decision at that level.

That makes the `Database` more flexible yet you can still easily do the if check if you want to on top of it.